backend/src/main/java/com/capstone/moneytree/controller/UserController.java

```diff
@@ -41,75 +50,56 @@ public UserController(UserService userService) {
       return users;
    }
 
-    /**
-     * A POST method that receives a user JSON object and registers it
-     * @param user The JSON object body
-     * @return A proper response with message
-     */
-    @PostMapping("/create-user")
-    public ResponseEntity<User> createUser(@RequestBody User user) {
-        ResponseEntity<User> response = null;
-        HttpStatus status = validateUserCreation(user);
-
-        if (status == HttpStatus.OK) {
-            User createdUser = userService.createUser(user);
-            response = ResponseEntity.ok(createdUser);
-        }
-
-        return response;
-    }
-
-    /**
-     * A method that updates a user with a new Alpaca key. This should be done only once at beginning
-     * @param id The user ID sent from the frontend
-     * @param key The Alpaca API key sent from the frontend
-     * @return The new updated user from the database
-     */
-    @PostMapping("/register-alpaca-key/{id}/{key}")
-    public ResponseEntity<User> registerAlpacaApiKey(@PathVariable Long id, @PathVariable String key) {
-        // TODO: Refactor this into a validation class validateString method to validate strings & message class too
-        if (StringUtils.isBlank(key)) {
-            throw new IllegalArgumentException();
-        }
-        User updatedUser = userService.registerAlpacaApiKey(id, key);
-
-        if (updatedUser == null) {
-            throw new EntityNotFoundException(ExceptionMessage.ENTITY_NOT_FOUND.getMessage());
-        }
-
-        return ResponseEntity.ok(updatedUser);
-    }
-
-    HttpStatus validateUserCreation(User user) {
-        String email = user.getEmail();
-        String username = user.getUsername();
-        String password = user.getPassword();
-        String firstName = user.getFirstName();
-        String lastName = user.getLastName();
-
-        // TODO: Refactor this into a validation class validateString method to validate strings & message class too
-        if (StringUtils.isBlank(email) || StringUtils.isBlank(username) || StringUtils.isBlank(password) ||
-                StringUtils.isBlank(firstName) || StringUtils.isBlank(lastName)) {
-            throw new IllegalArgumentException();
-        } else if (userService.userExists(email, username)) {
-            throw new UserAlreadyExistsException(ExceptionMessage.USER_ALREADY_EXISTS.getMessage());
-        }
-
-        return HttpStatus.OK;
-    }
-
-   @GetMapping("/users/{id}")
+   /**
+    * A POST method that receives a user JSON object and registers it
+    *
+    * @param user The JSON object body
+    * @return A proper response with the URI of the newly created user
+    */
+   @PostMapping("/create-user")
+   public ResponseEntity<User> createUser(@RequestBody User user) {
+      User createdUser = userService.createUser(user);
+
+      URI userURI = ServletUriComponentsBuilder.fromCurrentContextPath().
```

Comment:
Can you explain what this line does?

Reply:
So based on HTTP protocol, when a resource is created, we should return 201 Created with a header that contains the url of the resource. Some client will not support receiving 200 when they create a resource. It is just a matter of following the protocol